### PR TITLE
Integrate --relay-chain-rpc-url flag

### PR DESCRIPTION
**Pull request by Chralt98**
### What does it do?

Fixes https://github.com/zeitgeistpm/zeitgeist/issues/1010

Allows to specify a remote relay chain instead of the internal default relaychain.

### What important points should reviewers know?

Bootnodes branch had to be used otherwise `cumulus-relay-chain-minimal-node` had the specified conflicting internal packages from paritytech instead of our fork.

### Is there something left for follow-up PRs?

From polkadot-v0.9.36 and on, the cumulus_client_service crate has `build_relay_chain_interface` internally. So we can remove our `build_relay_chain_interface` then.

### Are there relevant PRs or issues?

https://github.com/PureStake/moonbeam/pull/1388

https://github.com/PureStake/moonbeam/pull/2042

### CODE REVIEW

File: node/src/service/service_parachain.rs [hunk 1]
```diff
@@ -231,6 +236,32 @@ where
     })
 }
 
+// TODO `build_relay_chain_interface` will be included in polkadot-v0.9.36 in cumulus-client-service here: https://github.com/paritytech/cumulus/commit/babf73bbc6ade358db105580b68aacb91550faa5#diff-52df5aabcc0b97cfe43ce4b88b9dcf52de661934e481f6ae935435f4941b1639R224-R252
+/// Build a relay chain interface.
+/// Will return a minimal relay chain node with RPC
+/// client or an inprocess node, based on the [`CollatorOptions`] passed in.
```
Comment (maltekliemann) [inline node/src/service/service_parachain.rs:242]: Do we have an issue for this?

**Review by maltekliemann — DISMISSED** [review]

**Review by sea212 — CHANGES_REQUESTED** [review]
Works as expected.

After the removal of the unused rpc flags in the standalone client this PR is ready to merge from my perspective.

File: node/src/service/service_parachain.rs [hunk 2]
```diff
@@ -231,6 +236,32 @@ where
     })
 }
 
+// TODO `build_relay_chain_interface` will be included in polkadot-v0.9.36 in cumulus-client-service here: https://github.com/paritytech/cumulus/commit/babf73bbc6ade358db105580b68aacb91550faa5#diff-52df5aabcc0b97cfe43ce4b88b9dcf52de661934e481f6ae935435f4941b1639R224-R252
```
Comment (Chralt98) [inline node/src/service/service_parachain.rs:239]:
```suggestion
// TODO(#1040) `build_relay_chain_interface` will be included in polkadot-v0.9.36 in cumulus-client-service here: https://github.com/paritytech/cumulus/commit/babf73bbc6ade358db105580b68aacb91550faa5#diff-52df5aabcc0b97cfe43ce4b88b9dcf52de661934e481f6ae935435f4941b1639R224-R252
```

File: node/src/service/service_parachain.rs (hunk 1 above)
Comment (Chralt98) [reply node/src/service/service_parachain.rs]: https://github.com/zeitgeistpm/zeitgeist/issues/1040

https://github.com/zeitgeistpm/zeitgeist/pull/1037/commits/28071b5975935bf4df049bc07804d208c612f328

File: node/src/service/service_parachain.rs [hunk 3]
```diff
@@ -231,6 +236,32 @@ where
     })
 }
 
+// TODO(#1040) `build_relay_chain_interface` will be included in polkadot-v0.9.36 in cumulus-client-service here: https://github.com/paritytech/cumulus/commit/babf73bbc6ade358db105580b68aacb91550faa5#diff-52df5aabcc0b97cfe43ce4b88b9dcf52de661934e481f6ae935435f4941b1639R224-R252
```
Comment (maltekliemann) [inline node/src/service/service_parachain.rs:239]: Wasn't this taken from PureStake, as well?

**Review by maltekliemann — APPROVED** [review]
Seems fine to me except that the licensing question needs to be answered and formatting be fixed.

File: node/src/service/service_parachain.rs (hunk 3 above)
Comment (Chralt98) [reply node/src/service/service_parachain.rs:239]: No, I could find it in the cumulus repository at a later version.

PureStake just gave me a hint that cumulus already implemented it.

**Review by sea212 — APPROVED** [review]

File: node/src/service/service_parachain.rs (hunk 3 above)
Comment (maltekliemann) [reply node/src/service/service_parachain.rs:239]: I mean, isn't this just this: https://github.com/PureStake/moonbeam/blob/84c39610fcefbbfc57054ccbb8f5a10e9c2cb63d/node/service/src/lib.rs#L501-L528 with the `let` statement replaced with a `match` expression? Not sure if we need to add copyright here.

**Review by maltekliemann — APPROVED** [review]

File: node/src/service/service_parachain.rs (hunk 3 above)
Comment (Chralt98) [reply node/src/service/service_parachain.rs:239]: Yes it is the same, but also from [here](https://github.com/paritytech/cumulus/commit/babf73bbc6ade358db105580b68aacb91550faa5#diff-52df5aabcc0b97cfe43ce4b88b9dcf52de661934e481f6ae935435f4941b1639R224-R252).

**Pull request merged**